### PR TITLE
add test with -OO

### PR DESCRIPTION
**Pull request by linchiwei123**
## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way

This simple test may prevent something like issue #17970

### CODE REVIEW

File: lib/matplotlib/tests/test_matplotlib.py [hunk 1]
```diff
@@ -42,3 +42,20 @@ def parse(key):
...
+
+
+def test_importable_with__OO():
+    """
+    When using -OO or export PYTHONOPTIMIZE=2, docstrings are discarded,
+    this simple test may prevent something like issue #17970.
+    """
+    program = (
+        "import numpy as np; "
+        "import numpy.ma as ma; "
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_matplotlib.py:54]: Don't really need these.

File: lib/matplotlib/tests/test_matplotlib.py [hunk 2]
```diff
@@ -42,3 +42,20 @@ def parse(key):
...
+        "import matplotlib as mpl; "
+        "import matplotlib.pyplot as plt; "
+        "import matplotlib.cbook as cbook; "
+        "import matplotlib.patches as mpatches"
+    )
+    cmd = '{} -OO -c "{}"'.format(sys.executable, program)
+    assert subprocess.call(cmd, shell=True) == 0
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_matplotlib.py:61]: Use a list, not a string.
```suggestion
    cmd = [sys.executable, '-OO', '-c', program]
    assert subprocess.call(cmd) == 0
```

File: .gitignore [hunk 3]
```diff
@@ -104,3 +104,6 @@ lib/z.lib
 #########################
 lib/matplotlib/backends/web_backend/node_modules/
 lib/matplotlib/backends/web_backend/package-lock.json
+
+
+venv/
```
Comment (QuLogic) [inline .gitignore:109]: This change is unrelated to this PR.

Comment (linchiwei123) [reply .gitignore]: need rebase?

Comment (QuLogic) [reply .gitignore]: Or we can squash merge?

Comment (linchiwei123) [reply .gitignore]: yes!
